### ISSUE TITLE
How to code your own greeting time slots

### ISSUE THREAD

**Issue by Binaryash** [evancohen/smart-mirror#274]
How to customize the greetings ::
First, open the file ...  /home/pi/smart-mirror/js/controller.js
To define your own time frames you need to modify this file.  If you compare it to the default file, you will see that I've changed it slightly to add an afternoon slot. 
You can add whatever time slots you wish
Look for this section in the file >>

var greetingUpdater = function () {
                if(!Array.isArray(config.greeting) && typeof config.greeting.midday != 'undefined') {
                    var hour = moment().hour();
                    var greetingTime = "midday";

                    if (hour > 6 && hour < 12) {
                        greetingTime = "morning";
                    } else if (hour >= 14 && hour < 18) {
                        greetingTime = "afternoon";
                    } else if (hour >= 18 && hour < 22) {
                        greetingTime = "evening";
                    } else if (hour >= 22 || hour < 6) {
                        greetingTime = "night";
                    }
                    var nextIndex = Math.floor(Math.random() * config.greeting[greetingTime].length);
                    var nextGreeting = config.greeting[greetingTime][nextIndex]
                    $scope.greeting = nextGreeting;
                }else if(Array.isArray(config.greeting)){
                    $scope.greeting = config.greeting[Math.floor(Math.random() * config.greeting.length)];
                }
            };
            greetingUpdater();
            $interval(greetingUpdater, 120000);
		
This is the code I added::
else if (hour >= 14 && hour < 18) {
                        greetingTime = "afternoon";
                    }	
		
			
			
The next thing you need to do is to add the greetings you wish to display to reflect the time slot you have added. Open the file  home/pi/smart-mirror/config.js  and look for the following section>>

 greeting : {
       night: ["Bed?", "zZzzZz", "Time to sleep"],
       morning: ["Good morning.."],
       midday: ["Hey! FOOD !", "Lunch"],
       afternoon: ["Nap?!", "Afternoon"],
       evening: ["Evening", " Any films?", "Not long till bed"]
    },

You can see the greeting I have added for the afternoon. Any variable you add to the controller file can be displayed here.

So, in a nutshell, add your time slot to the controller.js file. Then define the greeting to be displayed in the config.js file.
